docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,152 @@
+I.e. Extract the current status of the BMH, store it in an annotation, create the BMH on the new hub:
+```
+# Save the status in json format to a file
+kubectl get bmh <name-of-bmh> -o jsonpath="{.status}" > status.json
+# Save the BMH and apply the status annotation to the saved BMH.
+kubectl -n metal3 annotate bmh <name-of-bmh> \
+  baremetalhost.metal3.io/status="$(cat status.json)" \
+  --dry-run=client -o yaml > bmh.yaml
+```
+This flow ensures that the BMH on the new hub has status (BMO applies the status using the annotation). Hence, as the BMH should be in state 'Provisioned', the BMH reconcile is avoided.
+I.e. prevent destruction of restored spoke clusters.
+
+##### Suggested solution for automating the flow (make it transparent to the user):
+
+In assisted-service -> `bmh_agent_controller`:
+* If a BMH is in 'Provisioned' state (and, doesn't have `status`+`paused` annotations):
+  * Extract BMH's status and add `baremetalhost.metal3.io/status` to the object metadata.
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+* If a BMH is missing 'Status' (and, has `status`+`paused` annotation):
+  * To restore the original Status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+      * The `status` and `paused` annotation will be restored (see previous bullet). 
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * The entire inventory exists in Agent.Status. But, need to ensure it exists also post restore. If not, we can consider adding an annotation for it (i.e something as `status` in BMH).
+   * Note: some of the inventory data can also be restored from existing annotations.
```

Comment:
Is there any reason to use these annotations over the inventory in the status?

Reply:
Just in case we discover there's an issue with using the Agent status.

Reply:
Also, shouldn't we consider a scenario in which Agents are applied without status?

Reply:
Right, I forgot the DR doesn't apply status by default.

Maybe we can check if we can enable status out of the box for agents rather than in the `Restore` CR?